pkg/piecestore/psserver/psdb/psdb.go

```diff
@@ -245,15 +248,46 @@ func (db *DB) WriteBandwidthAllocToDB(rba *pb.Order) error {
 	// If the agreements are sorted we can send them in bulk streams to the satellite
 	t := time.Now()
 	startofthedayunixsec := time.Date(t.Year(), t.Month(), t.Day(), 0, 0, 0, 0, t.Location()).Unix()
-	_, err = db.DB.Exec(`INSERT INTO bandwidth_agreements (satellite, agreement, signature, uplink, serial_num, total, max_size, created_utc_sec, expiration_utc_sec, action, daystart_utc_sec) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)`,
+	_, err = db.DB.Exec(`INSERT INTO bandwidth_agreements (satellite, agreement, signature, uplink, serial_num, total, max_size, created_utc_sec, status, expiration_utc_sec, action, daystart_utc_sec) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)`,
 		rba.PayerAllocation.SatelliteId.Bytes(), rbaBytes, rba.GetSignature(),
 		rba.PayerAllocation.UplinkId.Bytes(), rba.PayerAllocation.SerialNumber,
-		rba.Total, rba.PayerAllocation.MaxSize, rba.PayerAllocation.CreatedUnixSec,
+		rba.Total, rba.PayerAllocation.MaxSize, rba.PayerAllocation.CreatedUnixSec, "UNSENT",
 		rba.PayerAllocation.ExpirationUnixSec, rba.PayerAllocation.GetAction().String(),
 		startofthedayunixsec)
 	return err
 }
 
+// DeleteBandwidthAllocationPayouts delete paid and/or old payout enteries based on days old
+func (db *DB) DeleteBandwidthAllocationPayouts() error {
+	defer db.locked()()
+
+	//@TODO make a config value for older days
+	t := time.Now().Add(time.Hour * 24 * -90).Unix()
+	_, err := db.DB.Exec(`DELETE FROM bandwidth_agreements WHERE created_utc_sec < ?`, t)
+	if err == sql.ErrNoRows {
+		err = nil
+	}
+	return err
+}
+
+// UpdateBandwidthAllocationStatus update the bwa payout status
+func (db *DB) UpdateBandwidthAllocationStatus(serialnum string, status pb.AgreementsSummary_Status) (err error) {
+	defer db.locked()()
+
+	switch status {
+	case pb.AgreementsSummary_FAIL:
```

Comment:
Status `FAIL` is not equal to `REJECTED`. `FAIL` should be set when storage node tried sending agreement to satellite but for some reason it failed (failed connection, old satellite). Such `FAIL` should be sent once again with next iteration. Status `REJECTED` is a state when the satellite explicitly stated that this agreement is wrong (malformed signature, corrupted data, etc.) and such agreement shouldn't be sent once again.